doc/contributing/coding-conventions.chapter.md

```diff
@@ -512,3 +512,73 @@ If you do need to do create this sort of patch file, one way to do so is with gi
     ```ShellSession
     $ git diff > nixpkgs/pkgs/the/package/0001-changes.patch
     ```
+
+## Package tests {#sec-package-tests}
+
+Tests are important to ensure quality and make reviews and automatic updates easy.
+
+Nix package tests are a lightweight alternative to [NixOS module tests](https://nixos.org/manual/nixos/stable/#sec-nixos-tests). They can be used to create simple integration tests for packages while the module tests are used to test services or programs with a graphical user interface on a NixOS VM. Unittests that are included in the source code of a package should be executed in the `checkPhase`.
+
+### Writing package tests {#ssec-package-tests-writing}
+
+This is an example using the `phoronix-test-suite` package with the current best practices.
+
+Add the tests in `passthru.tests` to the package definition like this:
```

Comment:
i would like to link `checkPhase` and `passthru.tests` inside the document, but i don't know how and if its worth adding now when the build pipeline changes soon and it might brake

Reply:
Good idea, current level of interlinking in the manual leaves a lot to be desired.

If you just use the anchor name, it will work now and we will migrate it:

```suggestion
Add the tests in [`passthru.tests`](#var-meta-tests) to the package definition like this:
```

Reply:
I rember the first time I saw `passthru`, I went like: "?!?!!?!? (It's not even an english word)"

Maybe one small phrase to introduce that concept here would be in order

> By convention, we use the `passthru` attribute which allows us to store arbitrary data, ...

(If that makes sense)

Reply:
@blaggacao yeah, that's true. there is reference documentation at https://nixos.org/manual/nixpkgs/stable/#sec-standard-meta-attributes